app/components/support_interface/reasons_for_rejection_dashboard_component.html.erb

```diff
@@ -0,0 +1,299 @@
+<h3 class="govuk-heading-m govuk-!-font-size-27 govuk-!-margin-top-8">Course full</h3>
```

Comment:
I feel like this could be a component in itself?

Maybe `RejectionReasonDashboardComponent` or something similar.

So you could just pass in what you need.

`RejectionReasonDashboardComponent.new(
   rejection_reason: 'course_full_y_n', 
   total_count:  total_rejection_count("course_full_y_n")
   last_month: current_month_rejection_count("course_full_y_n")
   before_this_month: "previous_rejection_count("course_full_y_n")
)`

You can then get the remaining stuff you need for IDs etc. by adding a few methods like

```
def section_id
  "#{split_and_remove_y_n}_section"
end

def header
  split_and_remove_y_n.join(' ').capitalise
end 

def split_and_remove_y_n
  rejection_reason.split('_').reject { |string| string == 'y' || string == 'n' }
end  
```
This ☝️ can be cleaner but hopefully, you get the idea 👍

Reply:
Sorry that looks really ugly in markdown